pkg/repository/v1manifest/manifest.go

```diff
@@ -258,18 +286,18 @@ func (manifest *Timestamp) SnapshotHash() FileHash {
 	return manifest.Meta[ManifestURLSnapshot]
 }
 
-// VersionedURL looks up url in the snapshot and returns a modified url with the version prefix
-func (manifest *Snapshot) VersionedURL(url string) (string, error) {
+// VersionedURL looks up url in the snapshot and returns a modified url with the version prefix, and that file's length.
+func (manifest *Snapshot) VersionedURL(url string) (string, *FileVersion, error) {
 	entry, ok := manifest.Meta[url]
```

Comment:
seems key of Meta if just a file name. like `index.json`

Reply:
we should first extract the filename from the url parameter?

Reply:
I think all fields which identify a file in any manifest should be a URL, not just a filename. The reason is that if in the future we add directories, then the filename is not enough.

Reply:
in https://github.com/pingcap-incubator/tiup/blob/master/doc/design/manifest.md#snapshotjson it's a filename,
also `tools/migrate` will generate as filename.
then we need to fix both